components/UserAction/index.js

```diff
@@ -0,0 +1,47 @@
+import styled from "styled-components";
+import {useSession, signIn, signOut} from "next-auth/react";
+import { useRouter } from "next/router";
+
+export default function UserAction() {
+  const { data: session } = useSession()
+  const router = useRouter();
+
+  return (
+    <>
+      {session
+        ?
+        <StyledUserActions>
+              Hello {session.user.name}
+          <StyledButton onClick={()=> signOut()}>Sign Out</StyledButton>
+        </StyledUserActions>
+        :
+        <StyledUserActions>
+              <StyledButton onClick={()=>(router.push('/auth'))}>Sign In</StyledButton>
```

Comment:
Ideally this would be a `<Link href="/auth">`. You should only use `router.push` if the target is not known at render time. I do understand though that you want the same styling, so it won't be the end of the world to keep it.